src/Shared/UnitTests/MockEngine.cs

```diff
@@ -485,5 +487,51 @@ public object UnregisterTaskObject(object key, RegisteredTaskObjectLifetime life
             _objectCache.TryRemove(key, out object obj);
             return obj;
         }
+
+        int runningTotal = 0;
+        Semaphore cpuCount;
+        public int RequestCores(int requestedCores)
+        {
+            cpuCount ??= Semaphore.OpenExisting(ResourceSemaphoreName);
+
+            int coresAcquiredBeforeMoreCoresGetAcquired = runningTotal;
```

Comment:
I think this would be clearer if you made it:
```
int coresAcquired = 0;
...
if (cpuCount.WaitOne(0))
{
    coresAcquired++;
}
...
runningTotal += coresAcquired;
return coresAcquired;